usercase/interactor/definition_interactor_test.go

```diff
@@ -0,0 +1,186 @@
+			term:           "a word that does not get records",
+			responseLength: 0,
+			error:          model.ErrNotFound{Term: "a word that does not get records"},
+			hasError:       true,
+		},
+		{
+			name:           "failure - json bad request",
+			term:           "laksdasjlk lakjsdjajs lakjsda",
+			responseLength: 0,
+			error:          errors.New("unexpected end of JSON input"),
+			hasError:       true,
+		},
+		{
+			name:           "failure - missing api key",
+			term:           "yo",
+			responseLength: 0,
+			error:          model.ErrMissingApiKey{},
+			hasError:       true,
+		},
+	}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			mockRepo := MockDefinitionsRepo{}
+			mockPresenter := MockDefinitionPresenter{}
+			mockRepo.On("GetDefinitionsByTerm", test.term).Return(test.response, test.error)
+
+			interactor := NewDefinitionInteractor(mockRepo, mockPresenter)
+			definitions, err := interactor.urbanDictionaryRepository.GetDefinitionsByTerm(test.term)
+			if test.hasError {
+				assert.EqualError(t, err, test.error.Error())
+			}
+			if !test.hasError {
+				assert.Nil(t, err)
+			}
+			if definitions != nil {
+				assert.Equal(t, test.responseLength, len(definitions.Definitions))
+			}
+		})
+	}
+}
+
+func TestGetFromCSV(t *testing.T) {
+	testcases := []struct {
+		name         string
+		term         string
+		response     *model.List
+		definitionId string
+		error        error
+	}{
+		{
+			name:         "success - definition found",
+			term:         "yeah",
+			definitionId: "1234566",
+			response: &model.List{
+				Definitions: []*model.Definition{
+					{
+						Definition: "yeah definition",
+						Permalink:  "www.urbanctionary.example",
+						ThumbsUp:   90,
+						Author:     "Orlando",
+						Word:       "the",
+						Defid:      1234566,
+						WrittenOn:  "2006-04-30T20:18:42.000Z",
+						Example:    "Hell [yeah]",
+						ThumbsDown: 29,
+					},
+				},
+			},
+		},
+		{
+			name:         "failure - definition not found",
+			term:         "buddy",
+			definitionId: "111111",
+			response: &model.List{
+				Definitions: []*model.Definition{
+					{
+						Definition: "buddy definition",
+						Permalink:  "www.urbanctionary.example",
+						ThumbsUp:   90,
+						Author:     "Orlando",
+						Word:       "the",
+						Defid:      1234566,
+						WrittenOn:  "2006-04-30T20:18:42.000Z",
+						Example:    "[The] movie",
+						ThumbsDown: 29,
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			mockRepo := MockDefinitionsRepo{}
+			mockPresenter := MockDefinitionPresenter{}
+			mockRepo.On("GetDefinitionById", test.definitionId).Return(test.response, test.error)
+
+			interactor := NewDefinitionInteractor(mockRepo, mockPresenter)
+			definitions, _ := interactor.urbanDictionaryRepository.GetDefinitionById(test.definitionId)
+			id, _ := strconv.Atoi(test.definitionId)
+
+			if definitions.Definitions[0].Defid == id {
```

Comment:
Why this redundant if statements? The assert.Equal will take care of validating the information, otherwise you're just trying to make the tests not fail when they should if some information is wrong.